### ISSUE TITLE
Support cloud storage (self chat) fully and consistently

### ISSUE THREAD

**Issue by alex3kov** [majn/telegram-purple#388]
I currently see 3 issues with it (I'm sending stuff to myself in Android app and then checking it on PC - Pidgin 2.12.0 and telegram-purple 1.3.1):  

1. Sent images are only displayed as low resolution thumbnails
2. Sent files are not received at all
3. When I send a message, it appears in Pidgin as shown below, but after I close it there's no way to bring it up again - it's not saved in Contacts list.
![self](https://cloud.githubusercontent.com/assets/3616242/26031692/9842bd48-3885-11e7-8c91-a5e94fee6547.png)

**Comment by BenWiederhake** [comment]
1. The downsampling applies only to the display, as far as I can see.  I just uploaded an image of size 1280x1024, and it was displayed smaller than that.  Right-clicking the image in Pidgin an selecting "Save to" allowed me to save the full image of size 1280x1024.  (But not bit-identical.  Huh.  But I guess that's Telegram's fault, not pidgin or tgp.)  The image seems to be as big as can fit into the window, and never grow.
2. I cannot reproduce that. When I upload a small `.txt`, I see it just fine:
![bildschirmfoto_2017-05-14_11-26-14](https://cloud.githubusercontent.com/assets/2690845/26032684/585e31da-3899-11e7-9381-bb5f9cefdee6.png)  
  Do you have any hints how to get into this situation, or what type of file it was, and how large?
3. You're right about "no way to bring it up again" and "not in Contacts list".  But that's just the counter-intuitive GUI of pidgin, and the messages are not lost!  Use a different client to "bring it up again", then in the conversation window click in the menu "Conversation > Add Buddy" (or similar; I use the German UI) to add it to the list.  You may have to restart pidgin to change the icon.

So in short:
1. Yes, but that's pidgin's fault, and the full-size image is there.
2. Cannot reproduce, please help me/us in finding out why this happens on your machine.
3. Just counter-intuitive GUI of pidgin.  See above how to make it do what you want.

**Comment by alex3kov** [comment]
Thanks for your feedback. Here's mine :)  
1. Point taken, if I save the image it does save in the source definition. But the plugin behaves differently for cloud storage (downsizes to thumbnail - [example](http://imgur.com/IvjgqnC)) and for regular chats (displays full size - [example](http://imgur.com/5WUJYfq)). So it's a) inconsistent and b) viewing image in self-chat requires extra step of saving the image to make out what's in it.  
2. Just tried 5 files using (clip icon) -> File: 72Kb pdf, 4.3Mb Opus, 495Kb jpg, 20Kb text file (.out extension), 20Kb text file (.txt extension). Only the image file went through to Pidgin, and it was displayed as image in chat, not linked like in your example. Other files did not go through at all.  
3. This worked, thanks. It doesn't display images in the log though (I don't know if it's supposed to or it's a Pidgin limitation of text-only logs).

**Comment by BenWiederhake** [comment]
1. Huh, that's very strange.  For me it behaves consistently like "full size".  Anyway, if you're able to store it in the source resolution, then telegram-purple provides the correct size, and everything else is done by Pidgin, including the display.  The only difference in the setup I can see is that in the screenshot you're not a "Buddy" of yourself.  So I expect that 3 also fixes 1.
2. Yep, that sounds like an actual bug in tgp.  Which version are you using?  The most recent relevant commit I can find is from 13th March, and is included in version 1.3.1.  If you're already running 1.3.1, please also try out dev-1.4.0 for good measure.
3. Images in the logs are a known issue, nobody knows how to do it.  Maybe related to #362.

**Comment by alex3kov** [comment]
1. Played a bit more with it, narrowed down to this glitch - if Pidgin chat window is small (not fullscreen) at the moment of receiving the image, then it is downsized. Restoring Pidgin chat window does not increase the visible size of the image - it remains small. However, if you send another image to yourself, then in now-maximized Pidgin chat window both images become full size. It doesn't seem to depend on (3).
2. I'm using 1.3.1 (see top post), will try dev-1.4.0 when I get some time to compile it.
3. Ok, that point is closed.

**Comment by BenWiederhake** [comment]
1. Strange, but that's pidgin's fault, and I see the same behavior.
2. If the issue persists: Saving `Hello World!` with final linebreak (so that it's 13 bytes) works for me.  
  ![bildschirmfoto_2017-05-15_19-11-04](https://cloud.githubusercontent.com/assets/2690845/26069739/4cb405d0-39a2-11e7-8d53-7f9104049e86.png)
  Does that also work for you in dev-1.4.0?  If not, can you send me a file (e.g., via Dropbox or Telegram) that doesn't show up for you?

**Comment by alex3kov** [comment]
Tried  dev-1.4.0 - all files are working! Thanks. I'll wait for release.

**Comment by majn** [comment]
To solve 3. use the (usually preinstalled) **history** plugin, or the **enhanced history** plugin.

**Comment by majn** [comment]
So I assume that all issues are solved? Please re-open if there's still anything left
